src/App.js

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import { Routes, Route } from "react-router-dom";
+import { Container } from "react-bootstrap";
+import "./style.css";
+
+import useTopics from "./hooks/useTopics";
+import Header from "./component/Header";
+import Footer from "./component/Footer";
+import Navibar from "./component/Navibar";
+import Home from "./component/Home";
+import Topics from "./component/Topics";
+import Article from "./component/Article";
+
+function App() {
+  const {topics} = useTopics();
+
+  const [search, setSearch] = useState("");
+
+  return (
+    <Container fluid>
+      
+      <Container className="main-container">
+        <Header></Header>
+      </Container>
+
+      <Navibar topics={topics} search={search} setSearch={setSearch}></Navibar>
+
+      <Container className="main-container">
+        <Routes>
+          <Route path="/" element={<Home />} />
+          <Route path="/topics/:slug"element={<Topics topics={topics}/>} />
```

Comment:
<Home /> and <Topics /> are functionally the same component. This could be dryer by using the same component and using it to render either all articles or articles by topic. The naming path of /topics in the URL is confusing as the user may expect a list of topics. /articles/football etc makes more sense as we would the expect a list of articles filtered by the topic of football.